lens-data/lens/private/syntax/srcloc.rkt

```diff
@@ -70,6 +77,49 @@ module+ test
 (define (source-location->list src)
   (build-source-location-list src))
 
+(define (replace-source-location old new)
+  (update-source-location old
+    #:source (source-location-source new)
+    #:line (source-location-line new)
+    #:column (source-location-column new)
+    #:position (source-location-position new)
+    #:span (source-location-span new)))
+
+(define source-location->srcloc-lens
+  (make-lens
+   source-location->srcloc
+   replace-source-location))
+
+(define source-location->list-lens
+  (make-lens
+   source-location->list
+   replace-source-location))
+
+(define source-location-source-lens
+  (make-lens
+   source-location-source
+   (update-source-location _ #:source _)))
+
+(define source-location-line-lens
+  (make-lens
+   source-location-line
+   (update-source-location _ #:line _)))
+
+(define source-location-column-lens
+  (make-lens
+   source-location-column
+   (update-source-location _ #:column _)))
+
+(define source-location-position-lens
+  (make-lens
+   source-location-position
+   (update-source-location _ #:position _)))
+
+(define source-location-span-lens
+  (make-lens
+   source-location-span
+   (update-source-location _ #:span _)))
```

Comment:
The `source-location-<x>-lens` definitions seem repetitive enough to warrant a macro, like:

```racket
(define-simple-macro (define-source-location-lenses [lens-id:id getter:expr kw:kw] ...)
  (begin (define id (make-lens getter kw)) ...))

(define-source-location-lenses
  [source-location-source-lens source-location-source #:source]
  [...] ...)
```